applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1879,15 +1926,240 @@ void TwoFluidNavierStokes<TElementData>::ComputeSplitting(
+    ModifiedShapeFunctions::Pointer pModifiedShapeFunctions)
+{
+    Matrix enr_neg_interp = ZeroMatrix(NumNodes, NumNodes);
+    Matrix enr_pos_interp = ZeroMatrix(NumNodes, NumNodes);
+
+    for (unsigned int i = 0; i < NumNodes; ++i){
+        if (rData.Distance[i] > 0.0){
+            enr_neg_interp(i, i) = 1.0;
+        } else{
+            enr_pos_interp(i, i) = 1.0;
+        }
+    }
+
+    // Call the Interface negative side shape functions calculator
+    pModifiedShapeFunctions->ComputeInterfaceNegativeSideShapeFunctionsAndGradientsValues(
+        rInterfaceShapeFunctionNeg,
+        rInterfaceShapeDerivativesNeg,
+        rInterfaceWeightsNeg,
+        GeometryData::GI_GAUSS_2);
+
+    // Call the Interface negative side normal functions calculator
+    pModifiedShapeFunctions->ComputeNegativeSideInterfaceAreaNormals(
+        rInterfaceNormalsNeg,
+        GeometryData::GI_GAUSS_2);
+
+    for (unsigned int gp = 0; gp < rInterfaceNormalsNeg.size(); ++gp){
+        const double normal_norm = norm_2(rInterfaceNormalsNeg[gp]);
+        rInterfaceNormalsNeg[gp] /= normal_norm;
+    }
+
+    // Compute the enrichment shape function values at the interface gauss points using the enrichment interpolation matrices
+    rEnrInterfaceShapeFunctionPos = prod(rInterfaceShapeFunctionNeg, enr_pos_interp);
+    rEnrInterfaceShapeFunctionNeg = prod(rInterfaceShapeFunctionNeg, enr_neg_interp);
+}
+
+template <>
+ModifiedShapeFunctions::Pointer TwoFluidNavierStokes< TwoFluidNavierStokesData<2, 3> >::ModifiedShapeFunctionsUtility(
+    const GeometryType::Pointer pGeometry,
+    const Vector& rDistances)
+{
+    auto p_modified_sh_func = Kratos::make_shared<Triangle2D3ModifiedShapeFunctions>(pGeometry, rDistances);
+    return p_modified_sh_func;
+}
+
+template <>
+ModifiedShapeFunctions::Pointer TwoFluidNavierStokes< TwoFluidNavierStokesData<3, 4> >::ModifiedShapeFunctionsUtility(
+        const GeometryType::Pointer pGeometry,
+        const Vector& rDistances)
+{
+    auto p_modified_sh_func = Kratos::make_shared<Tetrahedra3D4ModifiedShapeFunctions>(pGeometry, rDistances);
+    return p_modified_sh_func;
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::CalculateCurvature(
+        const Matrix& rInterfaceShapeFunctions,
+        Kratos::Vector& rInterfaceCurvature)
+{
+    GeometryType::Pointer p_geom = this->pGetGeometry();
+    const unsigned int n_gpt = rInterfaceShapeFunctions.size1();
+
+    rInterfaceCurvature.resize(n_gpt, false);
+
+    for (unsigned int gpt = 0; gpt < n_gpt; ++gpt){
+        double curvature = 0.0;
+        for (unsigned int i = 0; i < NumNodes; ++i){
+            curvature += rInterfaceShapeFunctions(gpt,i) * (*p_geom)[i].FastGetSolutionStepValue(CURVATURE);
+        }
+        rInterfaceCurvature[gpt] = curvature;
+    }
+}
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::SurfaceTension(
+    const double coefficient,
+    const Kratos::Vector& rCurvature,
```

Comment:
```suggestion
    const Kratos::Vector& rCurvatureVector,
```
Otherwise it can be misunderstood with a nodal reference value.